### DISCUSSION TITLE
Load data type description

### DISCUSSION THREAD

**Discussion by manuelValch** [eProsima/DDS-Record-Replay#123] [General]
Hello, 

Trying dds-record-replay at the moment and I would like to know if 
there is a way to send .idl files as input for the record so that I can have full data type description in the mcap file. 

To have like DdsRecorderCommand topic without turning on data type discovery:
```
[vagrant@vm-oracle8 mcap-cli-0.0.39]$ mcap list schemas ~/ddsrecord/2024-01-31_12-50-58_GMT_output.mcap 
id  name                encoding  data                       
1   DdsRecorderCommand  omgidl    struct DdsRecorderCommand  
{                          
string command;        
string args;           
};                         

2   DdsRecorderStatus   omgidl    struct DdsRecorderStatus   
{                          
string previous;       
string current;        
string info;           
};                         

3   message            omgidl 
```
 with idl like this:
```
struct message{
    unsigned long index;
    string message;
};

```

**Comment by rsanchez15** [comment]
Hi Manuel, 

This is something that we have been thinking about but right now this feature is not in our short time roadmap.

Also please send me an email to raul@eprosima.com so we can keep in touch and maybe discuss more about this new feature.

**Comment by manuelValch** [comment]
hello @rsanchez15 thanks for the answer, 
Do you know if there is a way to define in Recorder_Config.yaml the data types of my topics ? 

Currently going through the readthedoc.io but it is not clear to me.

**Reply by manuelValch** [reply]
Ok so enable -typeobject with fastddsgen now I am able to visualize all the data through FoxGlove

Closing this topic thanks
